modyn/supervisor/internal/utils/evaluation_result_writer.py

```diff
@@ -0,0 +1,85 @@
+import json
+import pathlib
+from abc import ABC, abstractmethod
+
+# pylint: disable=no-name-in-module
+from modyn.evaluator.internal.grpc.generated.evaluator_pb2 import EvaluationData
+from torch.utils.tensorboard import SummaryWriter
+
+
+class AbstractEvaluationResultWriter(ABC):
+    """
+    Abstract class used to write evaluation results to the evaluation directory
+    """
+
+    def __init__(self, pipeline_id: int, trigger_id: int, eval_directory: pathlib.Path):
+        self.pipeline_id = pipeline_id
+        self.trigger_id = trigger_id
+        self.eval_directory = eval_directory
+
+    @abstractmethod
+    def add_evaluation_data(self, dataset_id: str, dataset_size: int, evaluation_data: list[EvaluationData]) -> None:
+        """
+        Called whenever a metric results are available for a particular dataset.
+
+        Args:
+            dataset_id: the involved dataset.
+            dataset_size: the size (amount of samples) of the dataset.
+            evaluation_data: contains the metric results.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def store_results(self) -> None:
+        """
+        Called in the end to store the results.
+        """
+        raise NotImplementedError()
+
+    @staticmethod
+    @abstractmethod
+    def get_name() -> str:
+        """
+        Defines the name used in the pipeline config for this result writer
+        """
+        raise NotImplementedError()
+
+
+class JsonResultWriter(AbstractEvaluationResultWriter):
```

Comment:
Should I put these `ResultWriters` in separate files? Since they are rather short and simple I put them together.

Reply:
I understand they are short but sometimes if you don't know where what is defined it is helpful if we maintain a class-file mapping I think. So if it's not too big of an effort, please move it to separate files. Thanks!

Reply:
I see. It's also better if we want to extend the system at some point. Moved them in the next commit.